docs/api_reference/python-api-reference.md

```diff
@@ -24,18 +32,10 @@ SPDX-License-Identifier: MPL-2.0
 ## data types
 
 ```{eval-rst}
-.. autoclass:: power_grid_model.data_types.Dataset
-.. autoclass:: power_grid_model.data_types.SingleDataset
-.. autoclass:: power_grid_model.data_types.BatchDataset
-.. autoclass:: power_grid_model.data_types.DataArray
-.. autoclass:: power_grid_model.data_types.SingleArray
-.. autoclass:: power_grid_model.data_types.BatchArray
-.. autoclass:: power_grid_model.data_types.DenseBatchArray
-.. autoclass:: power_grid_model.data_types.SparseBatchArray
-.. autoclass:: power_grid_model.dataset_definitions.DatasetType
-.. autoclass:: power_grid_model.dataset_definitions.ComponentType
-.. autodata:: power_grid_model.typing.ComponentAttributeMapping
-   :annotation: ComponentAttributeMapping
+.. automodule:: power_grid_model.data_types
+   :undoc-members:
+   :show-inheritance:
+   :exclude-members: BatchList,NominalValue,RealValue,AsymValue,AttributeValue,Component,ComponentList,SinglePythonDataset,BatchPythonDataset,PythonDataset
```

Comment:
If we are not to expose these, should we think about moving them to a `_data_types.py` module or something along those lines?